### PR TITLE
ENG 3886: Save the script that is executed

### LINKED ISSUE #77: Save the script that is executed
Since we are using a subprocess to execute the benchmark, we should have a pretty clean way of generating a script that someone could run to get similar results locally. 

Let's surface that in the results for now + we can make an update to conbench to store + display that later

### PR DESCRIPTION
**Pull request by boshek**
Will close #77 once merged

This PR implements a `output_script` argument for `run_one` which defaults to TRUE and some tests. There is also some light cleanup around tests and documentation. Running this suite of benchmarks:

```r
run_benchmark(write_file, source = "nyctaxi_2010-01")
```

populates a "bm-scripts" directory with files that named similar to the results files except with an "rscript-" prefix. 
```bash
bm-scripts
└── write_file
    ├── rscript-lz4-1-feather-arrow_table-nyctaxi_2010-01.json
    ├── rscript-lz4-1-feather-data_frame-nyctaxi_2010-01.json
    ├── rscript-lz4-1-parquet-arrow_table-nyctaxi_2010-01.json
    ├── rscript-lz4-1-parquet-data_frame-nyctaxi_2010-01.json
    ├── rscript-lz4-10-feather-arrow_table-nyctaxi_2010-01.json
    ├── rscript-lz4-10-feather-data_frame-nyctaxi_2010-01.json
    ├── rscript-lz4-10-parquet-arrow_table-nyctaxi_2010-01.json
    ├── rscript-lz4-10-parquet-data_frame-nyctaxi_2010-01.json
    ├── rscript-snappy-1-parquet-arrow_table-nyctaxi_2010-01.json
    ├── rscript-snappy-1-parquet-data_frame-nyctaxi_2010-01.json
    ├── rscript-snappy-10-parquet-arrow_table-nyctaxi_2010-01.json
    ├── rscript-snappy-10-parquet-data_frame-nyctaxi_2010-01.json
    ├── rscript-uncompressed-1-feather-arrow_table-nyctaxi_2010-01.json
    ├── rscript-uncompressed-1-feather-data_frame-nyctaxi_2010-01.json
    ├── rscript-uncompressed-1-parquet-arrow_table-nyctaxi_2010-01.json
    ├── rscript-uncompressed-1-parquet-data_frame-nyctaxi_2010-01.json
    ├── rscript-uncompressed-10-feather-arrow_table-nyctaxi_2010-01.json
    ├── rscript-uncompressed-10-feather-data_frame-nyctaxi_2010-01.json
    ├── rscript-uncompressed-10-parquet-arrow_table-nyctaxi_2010-01.json
    └── rscript-uncompressed-10-parquet-data_frame-nyctaxi_2010-01.json
```

I am not entirely certain whether "bm-scripts" is best place or if that should be nested within "results" but I will wait for feedback on that.

### CODE REVIEW

File: tests/testthat/helper.R [hunk 1]
```diff
@@ -0,0 +1,6 @@
+library(jsonlite)
+
+wipe_results <- function() {
+  unlink(test_path("results/"), recursive = TRUE)
+  unlink(test_path("bm-scripts/"), recursive = TRUE)
```
Comment (alistaire47) [inline tests/testthat/helper.R:5]: This implies the scripts are part of the results, and should probably be in the same directory, no? Not sure what's ideal here, but maybe an `outputs` dir with `results` and `scripts` subdirs?

**Comment by alistaire47** [comment]
I'm confused why we're storing scripts in JSON. Do we want them to be machine-readable? Are plain R files less so? I get this avoids needing subdirectories for each benchmark to contain setup/run/teardown/whatever scripts, but that's much easier to navigate as a human, and not that hard to navigate as a computer.

I guess that reaction is because when I see a `scripts` directory that just contains a lot of JSON I get confused

**Comment by jonkeane** [comment]
Nice work! I'll have more substantive comments later (and maybe we can pair on this when we're together too...). But this is going in the right direction. 

I totally agree that it's weird to have these scripts in json, but we will (eventually) be sending them up to conbench (once conbench can accept those...) so json makes sense. I was originally thinking that we would actually put that json inside of the results json (so that, for example, you could get the lines of the script from the results object) as opposed to a separate json file — do you think that's doable?

**Comment by boshek** [comment]
> do you think that's doable

Very. That's a much cleaner idea. I think it also removes the need to have an argument for whether to include the script or not. Probably just easier to assume it will be there.

File: tests/testthat/helper.R (hunk 1 above)
Comment (boshek) [reply tests/testthat/helper.R]: removed now

File: tests/testthat/helper.R [hunk 2]
```diff
@@ -0,0 +1,3 @@
+library(jsonlite)
+
+wipe_results <- function() unlink(test_path("results/"), recursive = TRUE)
```
Comment (alistaire47) [inline tests/testthat/helper.R:3]: Do we want to be more targeted with how we use this? We could remove specific results when we need, or everything on teardown and/or setup, if we like. I worry we're going to have to go figure out where this is getting called and comment it out if we want to see the results for debugging.

Also jsonlite is not necessary anymore

File: tests/testthat/test-bm-row-group-size.R [hunk 3]
```diff
@@ -12,3 +12,5 @@ test_that("row_group_size benchmark runs", {
     "arrowbench_results"
   )
 })
+
+wipe_results()
```
Comment (alistaire47) [inline tests/testthat/test-bm-row-group-size.R:16]: I don't think this should be in some benchmarks but not all. If it's going to affect all, it should be at the top level where one call handles everything.

File: tests/testthat/helper.R (hunk 2 above)
Comment (boshek) [reply tests/testthat/helper.R:3]: Ok @jonkeane is doing this here: https://github.com/ursacomputing/arrowbench/pull/92

**Pull request merged**